haystack/components/generators/hugging_face_api.py

```diff
@@ -0,0 +1,215 @@
+        :param stop_words: An optional list of strings representing the stop words.
+        :param streaming_callback: An optional callable for handling streaming responses.
+        """
+
+        huggingface_hub_import.check()
+
+        if isinstance(api_type, str):
+            api_type = HFGenerationAPIType.from_str(api_type)
+
+        api_params = api_params or {}
+
+        if api_type == HFGenerationAPIType.SERVERLESS_INFERENCE_API:
+            model = api_params.get("model")
+            if model is None:
+                raise ValueError(
+                    "To use the Serverless Inference API, you need to specify the `model` parameter in `api_params`."
+                )
+            check_valid_model(model, HFModelType.GENERATION, token)
+            model_or_url = model
+        elif api_type in [HFGenerationAPIType.INFERENCE_ENDPOINTS, HFGenerationAPIType.TEXT_GENERATION_INFERENCE]:
+            url = api_params.get("url")
+            if url is None:
+                raise ValueError(
+                    "To use Text Generation Inference or Inference Endpoints, you need to specify the `url` parameter in `api_params`."
+                )
+            if not is_valid_http_url(url):
+                raise ValueError(f"Invalid URL: {url}")
+            model_or_url = url
+
+        # handle generation kwargs setup
+        generation_kwargs = generation_kwargs.copy() if generation_kwargs else {}
+        generation_kwargs["stop_sequences"] = generation_kwargs.get("stop_sequences", [])
+        generation_kwargs["stop_sequences"].extend(stop_words or [])
+        generation_kwargs.setdefault("max_new_tokens", 512)
+
+        self.api_type = api_type
+        self.api_params = api_params
+        self.token = token
+        self.generation_kwargs = generation_kwargs
+        self.streaming_callback = streaming_callback
+        self._client = InferenceClient(model_or_url, token=token.resolve_value() if token else None)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+
+        :returns:
+            A dictionary containing the serialized component.
+        """
+        callback_name = serialize_callable(self.streaming_callback) if self.streaming_callback else None
+        return default_to_dict(
+            self,
+            api_type=self.api_type,
+            api_params=self.api_params,
+            token=self.token.to_dict() if self.token else None,
+            generation_kwargs=self.generation_kwargs,
+            streaming_callback=callback_name,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "HuggingFaceAPIGenerator":
+        """
+        Deserialize this component from a dictionary.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["token"])
+        init_params = data.get("init_parameters", {})
```

Comment:
```suggestion
        init_params = data["init_parameters"]
```
Let's not silently hide a malformed dict/YAML.